### ISSUE TITLE
singleton

### ISSUE THREAD

**Issue by code34** [Aloee/oo_extdb2.Altis#7]
Hello :)

can you give me more informations about this part : 

private ["_savedProtocolDefault", "_savedProtocolList"];

_savedProtocolDefault = [uiNamespace getVariable _varNameDatabase, "defaultProtocol"] call BIS_fnc_getFromPairs;
_savedProtocolList = [uiNamespace getVariable _varNameDatabase, "protoList"] call BIS_fnc_getFromPairs;

MEMBER("defaultProtocolID", _savedProtocolDefault);
MEMBER("protoList", _savedProtocolList);

why do you save thoses variables in uinamespace variable ? Is there a specific link with extDB2 implementation ?

**Comment by code34** [comment]
Hi Aloee 

Any news about this subject ? :) During last days, i was blocked with the protocole function, and finaly resolved my problem yesterday.

I will publish the simplification code.

**Comment by code34** [comment]
extDB2 is not so easy to use, and is not user friendly : )) we have a lot of work to improve that

**Comment by Aloee** [comment]
Hello code34
I save protocol identifier in uiNamespace, just because extdb2 retains the connection while  is running the game / server. This is handy for development, so as not to restart the game each time. It is not critical in the future it will be possible to remove it, if you want to.
At the moment I have not been working on this project, I did not have enough time. I consulted and came to the conclusion that I should not use OOP approach to work with the database, because the performance at the expense of processing a very large code falls sharply, and the rate of data acquisition is the base for comfortable work with the database:

Example 1 :

---

...
PUBLIC FUNCTION("", "testSum") {
(1 + 2 + 3 + 4);
};
...

_t1 = diag_tickTime;
for "_i" from 0 to 1000 do {
"testSum" call _databaseConn;
};
diag_log format["test 1 time: %1", (diag_tickTime - _t1)];

---

Example 2 :

---

SimpleFuncTest = {
(1 + 2 + 3 + 4);
};

_t1 = diag_tickTime;
for "_i" from 0 to 1000 do {
[] call SimpleFuncTest;
};
diag_log format["test 2 time: %1", (diag_tickTime - _t1)];

---

Result :
10:42:58 "test 1 time: 0.185997"
10:42:58 "test 2 time: 0.00500107"

The performance is very important for me today.
This does not mean that it is necessary to forget about it. For those projects which do not require work with a large volume of data, I think our framework will useful at the expense of ease of use.
OOP approach is very suitable for development, and I hope that it will someday be realizable for Arma without significant loss of performance.
